### DISCUSSION TITLE
Is there any trade-off in using the keystore?

### DISCUSSION THREAD

**Discussion by igorlnunes** [Cyfrin/foundry-full-course-cu#2583] [Q&A] [answered]
Using the keystore from Foundry, it's possible to perform contract deployment and verification, and it's also more secure:

```bash
forge script script/DeployFundMe.s.sol:DeployFundMe --rpc-url $SEPOLIA_RPC_URL --account $ACCOUNT_KEYSTORE --broadcast --verify --etherscan-api-key $ETHERSCAN_API_KEY -vvvv
```

Why is the private key still used in the `.env` file as shown below? Although it's mentioned that utilizing a keystore is more secure.

```bash
forge script script/DeployFundMe.s.sol --rpc-url $SEPOLIA_RPC_URL --private-key $PRIVATE_KEY --broadcast --verify --etherscan-api-key $ETHERSCAN_API_KEY
```

Is there any trade-off in using the keystore? How is it done in production?

**Comment by EngrPips** [comment, marked as answer]
There are no trade-offs, I believe, but Patrick teaches working with `.env` before the Foundry keystore was even a feature at all, so it is part of the course.

**Reply by KaneK31** [reply]
So would you recommend following the course with the use of the env file, i'm unsure how I would use "cast send" as shown in video 22 of foundry  basics with the implementation he demonstrates in 21

**Reply by EngrPips** [reply]
You can follow along using `.env` but you should try to learn how to use `Foundry` keystore as well so that when you want to do a real deployment you can do it safely.
